src/settings/MatchProfiles/ViewMatchProfile.js

```diff
@@ -245,40 +245,46 @@ export class ViewMatchProfile extends Component {
-            qualifierValue: get(incomingMatchExpression, ['qualifier', 'qualifierValue']),
-            comparisonPart: this.getLabel(COMPARISON_PARTS, get(incomingMatchExpression, ['qualifier', 'comparisonPart'])),
           },
-        },
-        incomingRecordType: get(matchDetails, 'incomingRecordType', ''),
-        matchCriterion: this.getLabel(CRITERION_TYPES, get(matchDetails, 'matchCriterion', '')) || <NoValue />,
-        existingMatchExpression: {
-          fields: [
-            {
+          incomingRecordType: get(matchProfile, 'incomingRecordType', ''),
+          matchCriterion: this.getLabel(CRITERION_TYPES, get(item, 'matchCriterion', '')) || <NoValue />,
+          existingMatchExpression: {
+            fields: [{
               label: 'field',
               value: getFieldMatched(existingField, matchProfile.existingRecordType) || <NoValue />,
+            }],
+            dataValueType: this.getLabel(VALUE_TYPES, get(existingMatchExpression, 'dataValueType', '')) || <NoValue />,
+            qualifier: {
+              qualifierType: this.getLabel(QUALIFIER_TYPES, get(existingMatchExpression, ['qualifier', 'qualifierType'])),
+              qualifierValue: get(existingMatchExpression, ['qualifier', 'qualifierValue']),
+              comparisonPart: this.getLabel(COMPARISON_PARTS, get(existingMatchExpression, ['qualifier', 'comparisonPart'])),
             },
-          ],
-          dataValueType: this.getLabel(VALUE_TYPES, get(existingMatchExpression, 'dataValueType', '')) || <NoValue />,
-          qualifier: {
-            qualifierType: this.getLabel(QUALIFIER_TYPES, get(existingMatchExpression, ['qualifier', 'qualifierType'])),
-            qualifierValue: get(existingMatchExpression, ['qualifier', 'qualifierValue']),
-            comparisonPart: this.getLabel(COMPARISON_PARTS, get(existingMatchExpression, ['qualifier', 'comparisonPart'])),
           },
-        },
-        existingRecordType: get(matchDetails, 'existingRecordType', ''),
-      }],
+          existingRecordType: get(matchProfile, 'existingRecordType', ''),
+        };
+      }),
     };
+
     const existingRecordLabel = FOLIO_RECORD_TYPES[matchProfile.existingRecordType]
       ? <FormattedMessage id={FOLIO_RECORD_TYPES[matchProfile.existingRecordType].captionId} />
       : '';
     const incomingRecordLabel = INCOMING_RECORD_TYPES[matchProfile.incomingRecordType]
       ? <FormattedMessage id={INCOMING_RECORD_TYPES[matchProfile.incomingRecordType].captionId} />
       : '';
     const componentsProps = {
+      'section-incoming-field': { stateFieldValue: matchProfile.incomingRecordType },
+      'section-incoming-qualifier': { stateFieldValue: matchProfile.incomingRecordType },
+      'section-incoming-qualifier-part': { stateFieldValue: matchProfile.incomingRecordType },
+      // 'section-incoming-static-value-text': { stateFieldValue: staticValueType },
```

Comment:
If this comment is unused, remove it, please.

Reply:
It will be removed later as a part of states handling fix but not now.